### PR TITLE
RT3722 Add -preserve_dates to x509 app

### PR DESCRIPTION
**Pull request by tmshort**
Add the -preserve_dates dates option to preserve dates when signing
a certificate.

### CODE REVIEW

**Comment by richsalz** [comment]
+1  seems highly useful.

**Comment by levitte** [comment]
Mighty useful, but a bit on the late side for 1.1.0.

**Comment by mattcaswell** [comment]
Not for 1.1.0

**Comment by tmshort** [comment]
Rebased on top of master.

File: doc/man1/x509.pod [hunk 1]
```diff
@@ -121,6 +122,11 @@ to attempt to obtain a functional reference to the specified engine,
 thus initialising it if needed. The engine will then be set as the default
 for all available algorithms.
 
+=item B<-preserve_dates>
+
+When signing a certificate, preserve the "notBefore" and "notAfter" dates instead
+of adjusting them to current time and duration.
```
Comment (mattcaswell) [inline doc/man1/x509.pod]: This confused me a bit. Does this mean preserve the "notBefore" and "notAfter" dates from the CSR? Also, how does this option interacts with the "-days" option. What happens if you supply both?

Comment (tmshort) [inline doc/man1/x509.pod]: -days are ignored... I should state that.

Comment (richsalz) [inline doc/man1/x509.pod]: should probably be an error if both are provided.

**Comment by tmshort** [comment]
This has been updated.

**Comment by tmshort** [comment]
Updated to lastest master

**Comment by tmshort** [comment]
Moving the source of the PR to my own repo. Please reference #3673.

**Comment by tmshort** [comment]
Reopening as the original.

**Comment by tmshort** [comment]
Ping @openssl/committers, this still indicates that there are requested changes, but it's been updated.

**Comment by tmshort** [comment]
@mattcaswell requested changes were made

**Comment by mattcaswell** [comment]
@richsalz are you still ok with this?

**Comment by richsalz** [comment]
Yes I am still okay with this.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**